driver-core/src/main/java/com/datastax/driver/core/ControlConnection.java

```diff
@@ -579,6 +581,10 @@ private static void refreshNodeListAndTokenMap(Connection connection, Cluster.Ma
             broadcastAddresses.add(row.getInet("peer"));
             if (metadataEnabled)
                 allTokens.add(row.getSet("tokens", String.class));
+            String dseWorkload = row.getColumnDefinitions().contains("workload") ? row.getString("workload") : null;
```

Comment:
It looks like this seems to only set the workload and dse version for peers, not for the contact point itself.